rust/arrow/src/array/data.rs

```diff
@@ -229,7 +229,7 @@ pub struct ArrayData {
 
     /// The child(ren) of this array. Only non-empty for nested types, currently
     /// `ListArray` and `StructArray`.
-    child_data: Vec<ArrayDataRef>,
+    child_data: Vec<ArrayData>,
```

Comment:
Would the impact of the change be that the overall size of a nested `ArrayData` becomes slightly larger; as we're now only using `Arc` on `Buffer`?